Sources/StreamChat/Database/DTOs/AttachmentDTO.swift

```diff
@@ -121,146 +119,121 @@ extension NSManagedObjectContext: AttachmentDatabaseSession {
+            attachment = raw.asAnyAttachment
         }
-            
-        return ChatMessageAttachmentSeed(
-            localURL: localURL,
-            fileName: title,
-            type: AttachmentType(rawValue: type),
-            localState: localState
-        )
+
+        return attachment
     }
     
     /// Snapshots the current state of `AttachmentDTO` and returns its representation for used in API calls.
     /// It's possible to introduce custom attachment types outside the SDK.
     /// That is why `RawJSON` object is used for sending it to backend because SDK doesn't know the structure of custom attachment.
-    func asRequestPayload() -> RawJSON? { .create(fromDTO: self) }
-}
-
-private extension RawJSON {
-    static func create(fromDTO dto: AttachmentDTO) -> RawJSON? {
-        if let data = dto.data,
-           let rawJSON = try? JSONDecoder.default.decode(RawJSON.self, from: data) {
-            return rawJSON
-        } else {
+    func asRequestPayload() -> AttachmentRequestBody? {
+        guard let data = data else {
             log.error("Internal error. Unable to decode attachment `data` for sending to backend.")
             return nil
         }
+        return .init(type: attachmentType, data: data)
+    }
+}
+
+struct AttachmentRequestBody: Encodable {
+    private enum CodingKeys: String, CodingKey {
+        case type
+    }
+
+    let type: AttachmentType
+    let data: Data
+
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+//        let payload = try? JSONDecoder.default.decode(RawJSON.self, from: data)
+        try container.encode(type, forKey: .type)
+        try data.encode(to: encoder)
     }
 }
 
-private extension AttachmentDTO {
+extension AttachmentDTO {
     /// Helper decoding method that logs error only if object exists.
     /// Returns `nil` if `Data` for decoding is `nil`.
-    func decoded<T: Decodable>(
-        _ type: T.Type,
-        from data: Data?
-    ) -> T? {
-        if let data = data {
-            let object: T?
-            do {
-                object = try JSONDecoder.default.decode(type, from: data)
-                return object
-            } catch {
-                log.error(
-                    "Failed to decode attachment of type:\(type) with hash: <\(id)>, "
-                        + "falling back to ChatMessageCustomAttachment."
-                        + "Error: \(error)"
-                )
-                return nil
-            }
-        } else {
+    func payload<T: Decodable>(ofType type: T.Type = T.self) -> T? {
+        guard let data = data else { return nil }
+
+        do {
+            return try JSONDecoder.default.decode(type, from: data)
+        } catch {
+            log.error(
+                "Failed to decode attachment of type:\(type) with hash: <\(id)>, "
+                    + "falling back to ChatMessageCustomAttachment."
+                    + "Error: \(error)"
```

Comment:
nit: This is also possible:
```
log.error("""
Failed to decode attachment of type:\(type) with hash: <\(id)>, \
falling back to ChatMessageCustomAttachment. Error: \(error)
""")
```

Reply:
see more here https://github.com/GetStream/stream-chat-swift/pull/1014#discussion_r621119420